File: app-geekhome/src/main/java/eu/geekhome/rest/icons/IconController.java
package eu.geekhome.rest.icons;

import eu.geekhome.rest.PluginsManager;
import eu.geekhome.services.repository.IconDto;
import eu.geekhome.services.repository.Repository;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

@Path("icons")
public class IconController {

    private final Repository _repository;

    @Inject
    public IconController(PluginsManager pluginsManager) {
        _repository = pluginsManager.getRepositories().get(0);
    }

    @POST
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public long postIcon(IconDto iconDto) {
        return _repository.saveIcon(iconDto);
    }

    @PUT
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public void putIcon(IconDto iconDto) {
        _repository.updateIcon(iconDto);
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public List<IconDto> getAllIconCategories() {
        return _repository.getAllIcons();
    }

    @GET
    @Path("/{id}/raw")
    @Produces("image/svg+xml;charset=utf-8")
    public String getRaw(@PathParam("id") long id) {
        return _repository.getIcon(id).getRaw();
    }

    @DELETE
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public void deleteIcon(@PathParam("id") long id) {
        _repository.deleteIcon(id);
    }
}

File: app-geekhome/src/main/java/eu/geekhome/rest/icons/IconController.kt
package eu.geekhome.rest.icons

import eu.geekhome.data.Repository
import eu.geekhome.data.icons.IconDto
import javax.inject.Inject
import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("icons")
class IconController @Inject constructor(private val repository: Repository) {

    @POST
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    fun postIcon(iconDto: IconDto?): Long {
        return repository
                    .saveIcon(iconDto!!)
    }

    @PUT
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    fun putIcon(iconDto: IconDto?) {
        repository
            .updateIcon(iconDto!!)
    }

    @get:Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    @get:GET
    val allIconCategories: List<IconDto>
        get() = repository
                    .getAllIcons()

    @GET
    @Path("/{id}/raw")
    @Produces("image/svg+xml;charset=utf-8")
    fun getRaw(@PathParam("id") id: Long): String {
        return repository
                    .getIcon(id).raw
    }

    @DELETE
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    fun deleteIcon(@PathParam("id") id: Long) {
        repository
            .deleteIcon(id)
    }
}